src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -450,34 +450,21 @@ private bool AllFixed()
             return true;
         }
 
-        private void AddBound(TypeSymbol addedBound, HashSet<TypeSymbol>[] collectedBounds, TypeParameterSymbol methodTypeParameter)
+        private void AddBound(TypeSymbolWithAnnotations addedBound, HashSet<TypeSymbolWithAnnotations>[] collectedBounds, TypeSymbolWithAnnotations methodTypeParameterWithAnnotations)
         {
-            Debug.Assert(IsUnfixedTypeParameter(methodTypeParameter));
+            Debug.Assert(IsUnfixedTypeParameter(addedBound, methodTypeParameterWithAnnotations));
+
+            var methodTypeParameter = (TypeParameterSymbol)methodTypeParameterWithAnnotations.TypeSymbol;
             int methodTypeParameterIndex = methodTypeParameter.Ordinal;
 
             if (collectedBounds[methodTypeParameterIndex] == null)
             {
-                collectedBounds[methodTypeParameterIndex] = new HashSet<TypeSymbol>();
+                collectedBounds[methodTypeParameterIndex] = new HashSet<TypeSymbolWithAnnotations>(TypeSymbolWithAnnotations.EqualsComparer.Instance);
```

Comment:
Nit: I wasn't clear why we didn't need to pass a Comparer before, but we do now. Is that a matter of implementing `IEquatable`?

Reply:
Yes. However, I thought it might be better to keep comparison outside of `TypeSymbolWithAnnotations`, as a separate `IEqualityComparer<TypeSymbolWithAnnotations>` implementation, to require callers to be explicit about the particular comparison.